src/include/storage/write_ahead_log/log_manager.h

```diff
@@ -68,25 +98,112 @@ class LogManager {
   void Flush();
 
  private:
+  friend class LogConsumerTask;
+
+  bool run_log_manager_;
+
   // TODO(Tianyu): This can be changed later to be include things that are not necessarily backed by a disk
-  // (e.g. logs can be streamed out to the network for remote replication)
-  BufferedLogWriter out_;
+  //  (e.g. logs can be streamed out to the network for remote replication)
   RecordBufferSegmentPool *buffer_pool_;
 
   // TODO(Tianyu): Might not be necessary, since commit on txn manager is already protected with a latch
   common::SpinLatch flush_queue_latch_;
   // TODO(Tianyu): benchmark for if these should be concurrent data structures, and if we should apply the same
-  // optimization we applied to the GC queue.
+  //  optimization we applied to the GC queue.
   std::queue<RecordBufferSegment *> flush_queue_;
 
   // These do not need to be thread safe since the only thread adding or removing from it is the flushing thread
   std::vector<std::pair<transaction::callback_fn, void *>> commits_in_buffer_;
 
+  // System path for log file
+  const char *log_file_path_;
+  // This stores all the buffers the serializer or the log consumer threads use
+  std::vector<BufferedLogWriter> buffers_;
+  // This is the buffer the serializer thread will write to
+  BufferedLogWriter *filled_buffer_;
+  // The queue containing empty buffers which the serializer thread will use
+  common::ConcurrentBlockingQueue<BufferedLogWriter *> empty_buffer_queue_;
+  // The queue containing filled buffers pending flush to the disk
+  common::ConcurrentBlockingQueue<BufferedLogWriter *> filled_buffer_queue_;
+
+  // The log consumer task which flushes filled buffers to the disk
+  LogConsumerTask *log_consumer_task_ = nullptr;
```

Comment:
Do you need a direct reference to the thread task, or just a queue (which can be processed on the back end by multiple threads?)

Reply:
Currently we only support a single task, so having a pointer to the task works fine for now. Having multiple tasks would require more changes that are out of the scope of the PR.